libs/networking/http_helpers.ml

```diff
@@ -1,17 +1,26 @@
 open Cohttp
 open Cohttp_lwt_unix
+
 (* Below we separate the methods out by async (returns Lwt promise),
    and sync (runs async method in lwt runtime)
 *)
 (* This way we can use the async methods in the language server,
    and other places too
 *)
 
+(*****************************************************************************)
+(* Client *)
+(*****************************************************************************)
+
+(* Create a client reference so we can swap it out with a testing version *)
+let client_ref = ref (module Client : Cohttp_lwt.S.Client)
```

Comment:
first class modules!

Reply:
Yea, it was either this or functors and that seemed too heavy handed.